### DISCUSSION TITLE
Pre-build operations for VSCode Devcontainers?

### DISCUSSION THREAD

**Discussion by slobodaapl** [devcontainers/community#160] [Q&A]
I am creating an integrated development environment for our devs and I need to be able to run a script, say `my_script.sh` before the devcontainer starts building and opening.

I cannot use `initializeCommand` cause the `my_script.sh` would dynamically modify the `devcontainer.json` -- so this needs to be able to happen before that file is loaded by VSCode for use.

Is there any way to do this?

**Comment by bamurtaugh** [comment]
Thanks for opening. I don't think there's a spec-supported way for modifying the devcontainer.json dynamically, but cc @chrmarti in case I'm missing something.

**Reply by chrmarti** [reply]
This is currently not possible. We could consider reloading the devcontainer.json after running any `initializeCommand`.

@slobodaapl Which parts of the devcontainer.json do you need to generate dynamically?

**Reply by slobodaapl** [reply]
@chrmarti With our script it should be able to modify the whole `decontainer.json` depending on user needs, including listing forwarded ports, changing the options, or adding new mounts. If `devcontainer.json` was loaded after `initializeCommand` that would be fantastic.
